plugins/inputs/procstat/procstat.go

```diff
@@ -391,6 +445,46 @@ func (p *Procstat) SimpleFindPids(f PIDFinder) ([]PID, map[string]string, error)
 // execCommand is so tests can mock out exec.Command usage.
 var execCommand = exec.Command
 
+func (p *Procstat) supervisorPIDs() ([]string, map[string]map[string]string, error) {
+
+	out, err := execCommand("supervisorctl", "status", strings.Join(p.SupervisorUnit, " ")).Output()
+	if err != nil {
+		if !strings.Contains(err.Error(), "exit status 3") {
+			return nil, nil, err
+		}
+	}
+	lines := strings.Split(string(out), "\n")
+	// Get the PID, running status, running time and boot time of the main process:
+	// pid 11779, uptime 17:41:16
+	// Exited too quickly (process log may have details)
+	Mainpids := make(map[string]map[string]string)
```

Comment:
Please do not use variables starting with an uppercase letter! In golang those variables, function or struct names are _exported_ which is clearly wrong for a local variable
```suggestion
	mainPids := make(map[string]map[string]string)
```